plugin/evm/validators_normalizer_test.go

```diff
@@ -0,0 +1,88 @@
+package evm
+
+import (
+	"math"
+	"math/rand"
+	"reflect"
+	"testing"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidatorsNormalizer_ByEpoch(t *testing.T) {
+	var (
+		idsList []ids.ShortID
+		weights []uint64
+
+		normTestData     = testData{}
+		copyNormTestData = testData{}
+	)
+
+	numValidators := 6
+	var epochs = [2]uint64{0, 1}
+	for i := 0; i < numValidators; i++ {
+		idsList = append(idsList, randID())
+		weights = append(weights, rand.Uint64())
+
+		if i >= numValidators/len(epochs) {
+			normTestData[epochs[0]] = map[ids.ShortID]uint64{
+				idsList[i]: weights[i],
+			}
+			copyNormTestData[epochs[0]] = map[ids.ShortID]uint64{
+				idsList[i]: weights[i],
+			}
+			continue
+		}
+
+		normTestData[epochs[1]] = map[ids.ShortID]uint64{
+			idsList[i]: weights[i],
+		}
+		copyNormTestData[epochs[1]] = map[ids.ShortID]uint64{
+			idsList[i]: weights[i],
+		}
+	}
+
+	validatorsCache := NewValidatorsCache(normTestData, WithCacheSize(128))
+	for k, v := range normTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+	normalizer := NewValidatorsNormalizer(validatorsCache)
+
+	t.Run("NonExistingKey", func(t *testing.T) {
+		res, err := normalizer.ByEpoch(uint64(7))
+		require.NoError(t, err)
+		assert.Len(t, res, 0)
+	})
+
+	t.Run("ExistingKey", func(t *testing.T) {
+		for _, epoch := range epochs {
+			result, err := normalizer.ByEpoch(epoch)
+			require.NoError(t, err)
+
+			var weightedNTData = map[ids.ShortID]uint64{}
+			for id, _ := range result {
+				weightedNTData[id] = copyNormTestData[epoch][id]
+			}
+
+			weighted := calcWeightRatio(weightedNTData)
+			assert.Equal(t, len(weighted), len(result))
+			assert.True(t, reflect.DeepEqual(weighted, result))
```

Comment:
You can use [`assert.ElementsMatch`](https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch) for this I believe

Reply:
Why to use it since it is using reflect.DeepEqual itself also?

Reply:
Because it was made for this 😄 Instead of combining two methods (check if it's true that they are equal) we just use a single one that does the same thing clearly (check if they are equal)

Reply:
Okay :smile: